### DISCUSSION TITLE
Generating with redocly.yaml from files behind proxy

### DISCUSSION THREAD

**Discussion by vilola** [openapi-ts/openapi-typescript#2042] [Q&A]
Hi, I have redocly.yaml file to defined openapi files and its outputs to generate multiple type files for every service we have. Until now I manually fetched all open api files and then run openapi-typescript command, but my idea is to change it to get the api files from git/remote url instead. The problem is that our bitbucket is behind proxy. When I try to curl the api file with auth and proxy settings, it works file. Is there a way to tell openapi-ts or redocly to use proxy? The only mention about this I found was on redocly page to set global variables HTTP_PROXY and HTTPS_PROXY, but it did nothing. 

Thanks for helping!

**Comment by drwpow** [comment]
Yeah as you discovered, the fetching of schemas is handled by [Redocly Core](https://github.com/Redocly/redocly-cli/tree/main/packages/core), because resolution and “bundling” has to make choices on when to load remote documents. You’re right the Redocly Core library does support a little but not a lot. There are so many different ways that people protect schemas, and this library doesn’t support a lot of complexity in dealing with that. Also as a general rule we don’t this library to be concerned with your security at all, or influence any of those decisions.

Your solution of just fetching manually and storing in a temporary location for complex security is generally what we recommend.
